Tophat/Utilities/ArtifactUnpacker.swift

```diff
@@ -92,6 +96,25 @@ final class ArtifactUnpacker: Sendable {
 
 		return destinationURL
 	}
+
+	private func extractTarGzipArtifact(at url: URL) throws -> URL {
+		// Use tar command to extract .tar.gz files
+		let destinationFileName = url.fileRootForTarGz
+		let destinationURL = url.deletingLastPathComponent().appending(path: destinationFileName)
+
+		log.info("Extracting tar.gz artifact at \(url)")
+
+		// Create destination directory
+		try FileManager.default.createDirectory(at: destinationURL, withIntermediateDirectories: true)
+
+		// Extract using tar command: tar -xzf <file> -C <destination>
+		let tarCommand = TarCommand.extract(archiveUrl: url, destinationUrl: destinationURL)
```

Comment:
Is ZIPFoundation able to extract `tar.gz` archives?  Wondering if there is any existing functionality we could leverage rather than calling out to `tar` ourselves.